### PR TITLE
#1553 optimise dependencies

### LINKED ISSUE #1553: Remove polyfil package from net 7/8
### Describe the feature.
I want my application's and libraries to be as small as possible and the minimum amount of dependencies.

#### Is your feature related to a problem? 
Additional dependencies are being added to my projects.

#### Describe the requested feature

When the framework is able to natively provide a dependency ie System.Diagnostics.DiagnosticSource that should be utilised rather than an explicit dependency. This will be the case for the net 7+ TFM.
        
#### Describe alternatives you've considered
Accept the additional dependencies

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by thompson-tomo**
Have gone and eliminated polyfils packages where possible as in those cases the dependencies are already natively being provided by the framework.

Closes #1553

### CODE REVIEW

File: src/KubernetesClient.Classic/KubernetesClient.Classic.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;net48</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;net48;netstandard2.1;net6.0</TargetFrameworks>
```
Comment (tg123) [inline src/KubernetesClient.Classic/KubernetesClient.Classic.csproj:4]: why netstandard2.1 and net6 here?
classic will support netstandard2.0 only

to use net6, please use main sdk

File: src/KubernetesClient/KubernetesClient.csproj [hunk 2]
```diff
@@ -7,7 +7,6 @@
     </PropertyGroup>
 
     <ItemGroup>
-        <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="7.0.0" />
```
Comment (tg123) [inline src/KubernetesClient/KubernetesClient.csproj:10]: see #1534

Comment (WeihanLi) [reply src/KubernetesClient/KubernetesClient.csproj:10]: The `UpDownCounter` introduced in 7.0, `net6.0` framework reference does not have this reference, so we have to reference the package at least for `net6.0`, maybe a condition could be added like `Condition="'$(TargetFramework)'='net6.0'"`, safe to remove for `net8.0`

Comment (thompson-tomo) [reply src/KubernetesClient/KubernetesClient.csproj:10]: good pickup

File: src/KubernetesClient.Classic/KubernetesClient.Classic.csproj (hunk 1 above)
Comment (thompson-tomo) [reply src/KubernetesClient.Classic/KubernetesClient.Classic.csproj]: have removed the netstandard 2.1 as didn't provide benefit.

File: src/KubernetesClient/KubernetesClient.csproj (hunk 2 above)
Comment (WeihanLi) [reply src/KubernetesClient/KubernetesClient.csproj:10]: seemed fix not pushed

File: src/KubernetesClient.Classic/KubernetesClient.Classic.csproj [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;net48</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;net48;net6.0</TargetFrameworks>
```
Comment (tg123) [inline src/KubernetesClient.Classic/KubernetesClient.Classic.csproj:4]: could you please elaborate why net6 is added here?

**Pull request closed without merging**